src/Symfony/Component/DependencyInjection/Tests/Dumper/XmlDumperTest.php

```diff
@@ -287,6 +291,34 @@ public function testDumpHandlesEnumeration()
         $this->assertEquals(file_get_contents(self::$fixturesPath.'/xml/services_with_enumeration.xml'), $dumper->dump());
     }
 
+    // https://github.com/symfony/symfony/issues/48178
+    /**
+     * @requires PHP 8.1
+     * @dataProvider provideDefaultClasses
+     */
+    public function testDumpHandlesDefaultEnumeration($class, $expectedFile)
+    {
+        $container = new ContainerBuilder();
+        $container
+            ->register('foo', $class)
+            ->setPublic(true)
+            ->setAutowired(true)
+            ->setArguments([2 => true]);
+
+        (new AutowirePass())->process($container);
+
+        $dumper = new XmlDumper($container);
+
+        $this->assertSame(file_get_contents(self::$fixturesPath.'/xml/'.$expectedFile), $dumper->dump());
+    }
+
+    public function provideDefaultClasses()
```

Comment:
data providers must be static in recent phpunit versions
```suggestion
    public static function provideDefaultClasses()
```